DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -181,12 +181,14 @@ protected function flattenExtensionConfiguration(array $config)
         );
 
         foreach ($config as $extension => $options) {
-            foreach ($options as $key => $value) {
+            $optionsMap = array_intersect_key($options, $extensionMap);
+
+            foreach ($optionsMap as $key => $value) {
                 foreach ($value as $source) {
                     if (!isset($extensionMap[$key][$source])) {
                         $extensionMap[$key][$source] = array();
                     }
-                    array_push($extensionMap[$key][$source], $extension);
+                    $extensionMap[$key][$source][$extension]['priority'] = $options['priority'];
```

Comment:
Please change the phpdoc for the return value of this method, which is now completely different unless I'm mistaken